code/game/objects/items/control_wand.dm

```diff
@@ -0,0 +1,101 @@
+#define WAND_OPEN "Open Door"
+#define WAND_BOLT "Toggle Bolts"
+#define WAND_EMERGENCY "Toggle Emergency Access"
+
+/obj/item/weapon/door_remote
+	icon_state = "gangtool-white"
+	item_state = "electronic"
+	icon = 'icons/obj/device.dmi'
+	name = "control wand"
+	desc = "Remotely controls airlocks."
+	w_class = 1
+	var/mode = WAND_OPEN
+	var/region_access = 1 //See access.dm
+	var/obj/item/weapon/card/id/ID
+	var/wand_access = /datum/job/assistant //This is for access. See access.dm for which jobs give what access. Use "Captain" if you want the wand to work on all doors.
+
+/obj/item/weapon/door_remote/New()
+	..()
+	ID = new /obj/item/weapon/card/id
+	ID.access = get_region_accesses(region_access)
+
+/obj/item/weapon/door_remote/attack_self(mob/user)
+	switch(mode)
+		if(WAND_OPEN)
+			mode = WAND_BOLT
+		if(WAND_BOLT)
+			mode = WAND_EMERGENCY
+		if(WAND_EMERGENCY)
+			mode = WAND_OPEN
+	user << "Now in mode: [mode]."
+
+/obj/item/weapon/door_remote/afterattack(obj/machinery/door/airlock/D, mob/user)
+	if(!istype(D))
+		return
+	if(!(D.hasPower()))
+		user << "[D] has no power!"
+		return
+	if(!D.requiresID())
+		user << "[D]'s ID scan is disabled!"
+		return
+	if(D.check_access(src.ID))
+		switch(mode)
+			if(WAND_OPEN)
+				if(D.density)
+					D.open()
+				else
+					D.close()
+			if(WAND_BOLT)
+				if(D.locked)
+					D.unbolt()
```

Comment:
`unbolt()` does not have a check for power. The proc should be refactored to include it instead of the things that call it, but for now, you need to check for it (or move the checks from the wire and interface to the proc itself)n